com.ibm.streamsx.inet/impl/java/src/com/ibm/streamsx/inet/http/HTTPRequest.java

```diff
@@ -126,6 +135,12 @@ public HTTPResponse sendRequest(IAuthenticate auth) throws Exception {
 			if(entity != null)
 				post.setEntity(entity);				
 			req = post;
+			if (connectionTimeout > 0.0) {
```

Comment:
Is it possible to do this once, rather than on every request.

Create a HttpParams once and re-use it?